macros/HTMLRefTable.ejs

```diff
@@ -91,78 +91,86 @@ function hasCommonTag(A, B) {
+let tag;
+let badge;
+let withBadges = $1 || false;
+let TagFilters = {
+       include : typeof $0 === 'string' ? [lowerMe($0)]
                 : $0 && $0.include && Array.isArray($0.include) ? $0.include.map(lowerMe)
                 : [],
         exclude : $0 && $0.exclude && Array.isArray($0.exclude) ? $0.exclude.map(lowerMe)
                 : []
-    },
-    ForcedElements = $0 && $0.elements && Array.isArray($0.elements) ? $0.elements.map(lowerMe) : [],
-    HTMLRefBaseURL = "/docs/Web/HTML/Element",
-    HTMLLocalBaseURL = "/" + env.locale + HTMLRefBaseURL + "/",
-    HTMLDocPages   = page.subpagesExpand("/en-US" + HTMLRefBaseURL, 1),
-    HTMLTags       = [];
+};
+let ForcedElements = $0 && $0.elements && Array.isArray($0.elements) ? $0.elements.map(lowerMe) : [];
+let HTMLRefBaseURL = "/docs/Web/HTML/Element";
+let HTMLLocalBaseURL = "/" + env.locale + HTMLRefBaseURL + "/";
+let HTMLDocPages   = await page.subpagesExpand("/en-US" + HTMLRefBaseURL, 1);
+let HTMLTags       = [];
 
 
 // BUSINESS LOGIC
 // --------------
 
 // Get the relevant pages for HTML Elements based on the template input; include only
 // the part of the title that's actually the element name
-HTMLDocPages.forEach(function (page) {
-    var p;
-    var t = lowerMe(page.slug.split("/").pop(-1));
-    var tags = page.tags.map(lowerMe);
+for(let page of HTMLDocPages) {
+    let p;
+    let t = lowerMe(page.slug.split("/").pop(-1));
+    let tags = page.tags.map(lowerMe);
 
     if (ForcedElements.indexOf(t) > -1
     || ((TagFilters.include.length === 0 ||  hasCommonTag(tags, TagFilters.include))
     &&  (TagFilters.exclude.length === 0 || !hasCommonTag(tags, TagFilters.exclude)))){
-        p = wiki.getPage(HTMLLocalBaseURL + t);
+        p = await wiki.getPage(HTMLLocalBaseURL + t);
 
         HTMLTags.push({
             tagName      : t,
-            
-            // We need to special case the "Heading elements" article which stand for all hn elements.
-            tagLink      : t !== 'heading_elements' ? template("HTMLelement", [t])
-                         : ['h1','h2','h3','h4','h5','h6'].map(function (tg) { return template("HTMLelement", [tg]); }).join(', '),
-            
+            // We need to special case the "Heading elements" article
+            // which stand for all hn elements.
+            tagLink      : (t !== 'heading_elements'
+                               ? await template("HTMLelement", [t])
+                               : (await Promise.all([
+                                   template("HTMLelement", ['h1']),
+                                   template("HTMLelement", ['h2']),
+                                   template("HTMLelement", ['h3']),
+                                   template("HTMLelement", ['h4']),
+                                   template("HTMLelement", ['h5']),
+                                   template("HTMLelement", ['h6'])
+                               ])).join(', ')),
```

Comment:
Nicely done!

Reply:
Shouldn’t it be `template("HTMLElement")`?

Reply:
Template names are case-insensitive, so I just kept the capitalization that was already in use in this file.